roles/big-service/tasks/main.yml

```diff
@@ -0,0 +1,21 @@
+---
+
+- name: stopping the {{service_name}}
+  service: name={{service_name}} pattern=/app/{{service_name}}/{{script_name}} state=stopped
+  ignore_errors: True
+
+- name: delete old service directory
+  shell: rm -rf /app/{{service_name}}
+  ignore_errors: True
+
+- name: copy files and folders to remote machine
+  copy: src=./ dest=/app/{{service_name}}/ owner=root group=root mode=755
+
+- name: write service file
+  copy: content="start on filesystem\nexec /app/{{service_name}}/{{script_name}}" dest=/etc/init/{{service_name}}.conf owner=root group=root mode=755
```

Comment:
This is indeed the "simplistic" way of implementing this.
The problem is that upstart doesn't really know if the service is running or not (no PID file), and it can crash without it being started again (bad thing on production)
The way to solve this issue is using `supervisord` (http://supervisord.org/) or something similar.